src/librustc_mir/hair/cx/expr.rs

```diff
@@ -75,21 +75,21 @@ fn apply_adjustment<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
                                     -> Expr<'tcx> {
     let Expr { temp_lifetime, mut span, .. } = expr;
     let kind = match adjustment.kind {
-        Adjust::Pointer(PointerCast::ReifyFnPointer) => {
-            ExprKind::ReifyFnPointer { source: expr.to_ref() }
-        }
-        Adjust::Pointer(PointerCast::UnsafeFnPointer) => {
-            ExprKind::UnsafeFnPointer { source: expr.to_ref() }
+        Adjust::Pointer(PointerCast::Unsize) => {
+            if let ExprKind::Block { body } = expr.kind {
```

Comment:
you lost the comment about looking at the comment on `Adjust::Deref`

Reply:
Yeah I thought the comment was misplaced. Why do we mention about `Adjust::Deref` in `Adjust::Unsize`?

Reply:
It's about the comment in line 94. This code exists solely to improve diagnostics. Maybe pull it out into a function instead of duplicating the logic?

Reply:
Added a new function `adjust_span `for this.